File: src/tools/lizard/eval-repos/synthetic/java/module/Nested.java
package synthetic.module;

import java.util.ArrayList;
import java.util.List;
import java.util.Optional;
import java.util.concurrent.locks.ReadWriteLock;
import java.util.concurrent.locks.ReentrantReadWriteLock;

/**
 * Nested module demonstrating Java module patterns.
 */
public class Nested {

    /**
     * Represents a nested item.
     */
    public static class NestedItem {
        private final int id;
        private final String name;
        private final long createdAt;

        public NestedItem(int id, String name, long createdAt) {
            this.id = id;
            this.name = name;
            this.createdAt = createdAt;
        }

        public int getId() {
            return id;
        }

        public String getName() {
            return name;
        }

        public long getCreatedAt() {
            return createdAt;
        }
    }

    /**
     * Service for managing nested items.
     */
    public static class NestedService {
        private final List<NestedItem> items = new ArrayList<>();
        private final ReadWriteLock lock = new ReentrantReadWriteLock();

        public void add(NestedItem item) {
            lock.writeLock().lock();
            try {
                items.add(item);
            } finally {
                lock.writeLock().unlock();
            }
        }

        public Optional<NestedItem> find(int id) {
            lock.readLock().lock();
            try {
                return items.stream()
                        .filter(item -> item.getId() == id)
                        .findFirst();
            } finally {
                lock.readLock().unlock();
            }
        }

        public List<NestedItem> getAll() {
            lock.readLock().lock();
            try {
                return new ArrayList<>(items);
            } finally {
                lock.readLock().unlock();
            }
        }

        public int clear() {
            lock.writeLock().lock();
            try {
                int count = items.size();
                items.clear();
                return count;
            } finally {
                lock.writeLock().unlock();
            }
        }
    }

    /**
     * Interface for processing items.
     */
    public interface NestedProcessor {
        boolean process(NestedItem item);
    }

    /**
     * Async processor implementation.
     */
    public static class AsyncProcessor implements NestedProcessor {
        @Override
        public boolean process(NestedItem item) {
            return item.getId() > 0;
        }
    }
}

File: src/tools/lizard/eval-repos/synthetic/csharp/Module/Nested.cs
namespace Synthetic.Module;

/// <summary>
/// Nested module demonstrating namespace and import patterns.
/// </summary>
public class NestedService
{
    private readonly List<string> _items = new();

    public void Add(string item)
    {
        if (!string.IsNullOrEmpty(item))
        {
            _items.Add(item);
        }
    }

    public IEnumerable<string> GetAll() => _items.AsReadOnly();

    public string? Find(Func<string, bool> predicate) =>
        _items.FirstOrDefault(predicate);
}

public record NestedRecord(int Id, string Name, DateTime Created);

public interface INestedProcessor
{
    Task<bool> ProcessAsync(NestedRecord record, CancellationToken ct = default);
}

public class NestedProcessor : INestedProcessor
{
    public async Task<bool> ProcessAsync(NestedRecord record, CancellationToken ct = default)
    {
        await Task.Delay(10, ct);
        return record.Id > 0;
    }
}
